app/app.go

```diff
@@ -340,18 +356,49 @@ func NewOsmosisApp(
 
 			// // configure upgrade for gamm module's pool creation fee param add
 			// app.GAMMKeeper.SetParams(ctx, gammtypes.NewParams(sdk.Coins{sdk.NewInt64Coin("uosmo", 1)})) // 1 uOSMO
+			// // execute prop12. See implementation in
 			// prop12(ctx, app)
 			return vm, nil
 		})
 
 	app.UpgradeKeeper.SetUpgradeHandler(
-		"v5", func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		v5UpgradeName,
+		func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+			// Set IBC updates from {inside SDK} to v1
+			// https://github.com/cosmos/ibc-go/blob/main/docs/migrations/ibc-migration-043.md#in-place-store-migrations
+			app.IBCKeeper.ConnectionKeeper.SetParams(ctx, ibcconnectiontypes.DefaultParams())
+
 			totalLiquidity := app.GAMMKeeper.GetLegacyTotalLiquidity(ctx)
 			app.GAMMKeeper.DeleteLegacyTotalLiquidity(ctx)
 			app.GAMMKeeper.SetTotalLiquidity(ctx, totalLiquidity)
-			return vm, nil
+
+			// Set all modules "old versions" to 1.
+			// Then the run migrations logic will handle running their upgrade logics
+			fromVM := make(map[string]uint64)
+			for moduleName := range app.mm.Modules {
+				fromVM[moduleName] = 1
+			}
+			// override versions for _new_ modules as to not skip InitGenesis
+			fromVM[authz.ModuleName] = 0
+			fromVM[txfees.ModuleName] = 0
```

Comment:
I think these need to be deleted from the map from looking at the code for RunMigrations: https://github.com/cosmos/cosmos-sdk/blob/994219a4d2f467a0bd628d60d8670c1102edbe88/types/module/module.go#L396-L444 . I'm not sure how gaia's upgrade / default genesis setting works. Maybe all of their modules need no state initialization?

As it stands, InitGenesis won't be ran.

Also, they both get ran with their DefaultGenesis -- we probably need to change that.

I think we have to do `RunMigrations`, then override their params as we want.

Reply:
Fixed in latest commit